app/src/main/java/com/android/streetworkapp/ui/profile/AddFriend.kt

```diff
@@ -0,0 +1,114 @@
+    userViewModel: UserViewModel = viewModel(factory = UserViewModel.Factory)
+) {
+  // variable for outlined text
+  var id by remember { mutableStateOf("") }
+  // context for Toast
+  val context = LocalContext.current
+  // fake user ID (placeholder)
+  val uid = "user123"
+
+  Scaffold(
+      modifier = Modifier.testTag("addFriendScreen"),
+      topBar = {
+        // button to go back
+        TopAppBar(
+            title = { Text("Add a new friend", Modifier.testTag("addFriendTitle")) },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() }, Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            })
+      },
+      content = { padding ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(padding).testTag("AddFriendColumn"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(20.dp, Alignment.CenterVertically)) {
+              Image(
+                  painter = painterResource(id = R.drawable.place_holder),
+                  contentDescription = "profile picture",
+                  modifier = Modifier.size(200.dp))
+
+              // button to activate NFC (don't work)
+              Button(onClick = {}, modifier = Modifier.size(220.dp, 50.dp).testTag("NFCButton")) {
+                Text(text = "Activate NFC", fontSize = 17.sp)
+              }
+
+              // write friend ID
+              OutlinedTextField(
+                  value = id,
+                  onValueChange = { id = it },
+                  label = { Text("Friend ID") },
+                  placeholder = { Text("Add a friend with id") },
+                  modifier =
+                      Modifier.width(300.dp) // make the text field smaller
+                          .testTag("inputID"))
+
+              // Put the id inside the friend list of USER
+              Button(
+                  onClick = {
+                    if (id.isEmpty()) {
+                      // If id is null or empty, show a toast message to the user
+                      Toast.makeText(context, "ID cannot be empty.", Toast.LENGTH_SHORT).show()
+                    } else {
+                      // add the friend to the user firendlist
+                      userViewModel.addFriend(uid, id)
+                      Toast.makeText(context, "Friend request sent.", Toast.LENGTH_SHORT).show()
+                      /*try {
+                        userViewModel.addFriend(uid, id)
+                        Toast.makeText(context, "Friend request sent.", Toast.LENGTH_SHORT).show()
+                      } catch (e: ApiException) {
+                        Toast.makeText(context, "Invalid ID", Toast.LENGTH_SHORT).show()
+                      }*/
```

Comment:
This should be removed, it is against conventions to let junk code commented out.